src/barril/units/_quantity.py

```diff
@@ -62,7 +64,7 @@ def ObtainQuantity(unit, category=None, unknown_unit_caption=None):
             unit = odict((cat, unit_and_exp) for (cat, unit_and_exp) in zip(category, unit))
             category = None
 
-    if unit.__class__ == odict:
+    if isinstance(unit, collections.Mapping):
```

Comment:
Oh, I see that it defines a new ordered dict class... I think we don't do many uses of odict in units and we should probably use a regular dict as it already retains the order in latest python versions (and make sure we use the same types in etk too).

Also, I remember this was a performance-wise critical area of the code, so, we shouldn't use `isinstance` here (i.e.: just check for the `items` attr in the `unit` instead of doing `isinstance` if we want to accept any type -- although I think we still want to check for the `odict` exact type and not any mapping as a dict on py2 would be wrong).

Reply:
I will change to check `items` then.

Btw, why does the order matter when defining units?

Reply:
Because we want to keep units and categories in order -- i.e.: something as m3.days != days.m3.